src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -326,25 +326,15 @@
     public ResponseEntity<List<RevisionDTO>> getSubjectRevisions(
             @ApiParam Pageable pageable, @PathVariable String login) throws NotAuthorizedException {
         log.debug("REST request to get revisions for Subject : {}", login);
-        SubjectDTO subject = subjectService.getLatestRevision(login);
-        Page<RevisionDTO> page = revisionService.getRevisionsForEntity(pageable,
-                subjectMapper.subjectDTOToSubject(subject));
-        SubjectDTO blank = new SubjectDTO();
-        RadarToken token = getJWT(servletRequest);
-        // iterate the revisions, if we don't have read permission for the subject in a given
-        // revision, swap the subject with a 'blank' subject so no information gets leaked
-        page = page.map(rev -> token.hasPermissionOnSubject(SUBJECT_READ,
-                ((SubjectDTO) rev.getEntity()).getProject().getProjectName(),
-                ((SubjectDTO) rev.getEntity()).getLogin())
-                ? rev : rev.setEntity(blank));
-        // This stream returns true if all values are equal to blank. To prevent people with no
-        // access to the requested subject in any of the subject history's projects from gaining
-        // information on the subject history this way, we throw a NotAuthorized.
-        if (page.getContent().stream()
-                .map(RevisionDTO::getEntity)
-                .allMatch(blank::equals)) {
-            throw new NotAuthorizedException();
-        }
+        Subject subject = subjectService.findOneByLogin(login);
+        Project project = subject.getActiveProject()
+                .flatMap(p -> projectRepository.findOneWithEagerRelationships(p.getId()))
+                .orElse(null);
+
+        checkPermissionOnSubject(getJWT(servletRequest), SUBJECT_READ, project.getProjectName(),
```

Comment:
This will throw NPE if project is null. better to throw intentionally in .orElse. I’m not sure about the implementation of checkpermissiononsubject if the project name is null. In that case, only a user or client with global level access should be able to see the data.